src/pages/Governance/ProposalList/CreateProposalModal/ProposalWizard/index.tsx

```diff
@@ -0,0 +1,174 @@
+  if (matchProposalInfoStep) {
+    return 'proposal-info';
+  }
+  if (matchProposalDescriptionsStep) {
+    return 'proposal-descriptions';
+  }
+  if (matchProposalActionsStep) {
+    return 'proposal-actions';
+  }
+  if (matchProposalPreviewStep) {
+    return 'proposal-preview';
+  }
+
+  return 'proposal-create';
+};
+
+export const getPreviousStep = (
+  currentStep: ProposalWizardSteps,
+  proposalMode: 'file' | 'manual',
+): ProposalWizardSteps => {
+  switch (currentStep) {
+    case 'proposal-preview':
+      return proposalMode === 'manual' ? 'proposal-actions' : 'proposal-create';
+    case 'proposal-actions':
+      return 'proposal-descriptions';
+    case 'proposal-descriptions':
+      return 'proposal-info';
+    default:
+      return 'proposal-create';
+  }
+};
+
+const getNextStep = (currentStep: ProposalWizardSteps): ProposalWizardSteps => {
+  switch (currentStep) {
+    case 'proposal-info':
+      return 'proposal-descriptions';
+    case 'proposal-descriptions':
+      return 'proposal-actions';
+    default:
+      return 'proposal-preview';
+  }
+};
+
+const ProposalWizard: React.FC<ProposalWizardProps> = ({
+  setProposalMode,
+  handleImportProposal,
+  isCreateProposalLoading,
+  currentStep,
+}) => {
+  const { path } = useRouteMatch();
+  const history = useHistory();
+  const { t } = useTranslation();
+
+  const formikContext = useFormikContext<FormValues>();
+  const { errors } = formikContext;
+
+  const getErrorsByStep = useCallback(
+    (step: ProposalWizardSteps) => {
+      switch (step) {
+        case 'proposal-info':
+          return !!(errors.title || errors.description);
+        case 'proposal-descriptions':
+          return !!(
+            errors.forDescription ||
+            errors.abstainDescription ||
+            errors.againstDescription
+          );
+        case 'proposal-actions':
+          return !!errors.actions;
+        default:
+          return true;
+      }
+    },
+    [currentStep, JSON.stringify(errors)],
+  );
+
+  const buttonText =
+    currentStep === 'proposal-preview'
+      ? t('vote.createProposalForm.confirm')
+      : t('vote.createProposalForm.nextStep');
+
+  return (
+    <>
+      <Switch>
+        <Route path={`${routes.governance.path}/proposal-create`}>
+          <UploadOrManualProposal
+            onClickCreateManually={() => {
+              setProposalMode('manual');
+              history.push(`${path}/proposal-info`);
+            }}
+            onClickUploadFile={async file => handleImportProposal(file, formikContext)}
+          />
+        </Route>
+        <Route path={`${path}/proposal-info`}>
+          <ProposalInfo />
+        </Route>
+        <Route path={`${path}/proposal-descriptions`}>
+          <VotingDescriptions />
+        </Route>
+        <Route path={`${path}/proposal-actions`}>
+          <ActionAccordion />
+        </Route>
+        <Route path={`${path}/proposal-preview`}>
+          <ProposalPreview />
+        </Route>
```

Comment:
Let's add these routes to the `routing` [file](https://github.com/VenusProtocol/venus-protocol-interface/blob/18f5d7c790c9c75fb3611244e3cf581d528be72a/src/constants/routing.ts) and import them from there. This way you'll also be able to check what the current step is by using the path directly